go-controller/pkg/node/node.go

```diff
@@ -74,13 +74,40 @@ func clearOVSFlowTargets() error {
 	return nil
 }
 
-func setOVSFlowTargets() error {
-	if config.Monitoring.NetFlowTargets != nil {
-		collectors := ""
-		for _, v := range config.Monitoring.NetFlowTargets {
-			collectors += "\"" + util.JoinHostPortInt32(v.Host.String(), v.Port) + "\"" + ","
+// collectorsString joins all HostPort entry into a string that is acceptable as
+// target by the ovs-vsctl command. If an entry has an empty host, it uses the Node IP
+func collectorsString(node *kapi.Node, targets []config.HostPort) (string, error) {
+	if len(targets) == 0 {
+		return "", nil
+	}
+	var joined strings.Builder
+	for n, v := range targets {
+		if n == 0 {
+			joined.WriteByte('"')
+		} else {
+			joined.WriteString(`","`)
+		}
+		var host string
+		if v.Host != nil && len(*v.Host) != 0 {
+			host = v.Host.String()
+		} else {
+			var err error
+			if host, err = util.GetNodePrimaryIP(node); err != nil {
```

Comment:
if cluster is single-stack IPv6, or dual stack with Primary IP IPv6?

Do we reject IPv6 addresses somewhere in the parsing?

Reply:
from the documentation of the inet_open_active function [invoked here](https://github.com/openvswitch/ovs/blob/master/ofproto/collectors.c#L66), I will assume that OVS accepts IPv6 addresses encoded as `[ip]:port`

Reply:
Samples
```
sh-4.4# /usr/bin/ovs-vsctl --timeout=15 -- --id=@ipfix create ipfix targets='["[fe80::d060:9787:9e17:b8b1]:2055"]'  cache_active_timeout=60 obs_domain_id=123 obs_point_id=456   -- set bridge br-int ipfix=@ipfix
sh-4.4# ovs-vsctl find ipfix
_uuid               : a4271ac9-9a92-4640-835d-2e2c3dd62b7e
cache_active_timeout: 60
cache_max_flows     : []
external_ids        : {}
obs_domain_id       : 123
obs_point_id        : 456
other_config        : {}
sampling            : []
targets             : ["[fe80::d060:9787:9e17:b8b1]:2055"]

sh-4.4# /usr/bin/ovs-vsctl --timeout=15 -- --id=@netflow create netflow targets='["[fe80::d060:9787:9e17:b8b1]:2055"]' active_timeout=60 -- set bridge br-int netflow=@netflow
6e0c1e55-6c66-4eac-9489-7458ddb3ddac
sh-4.4# ovs-vsctl find netflow
_uuid               : 6e0c1e55-6c66-4eac-9489-7458ddb3ddac
active_timeout      : 60
add_id_to_interface : false
engine_id           : []
engine_type         : []
external_ids        : {}
targets             : ["[fe80::d060:9787:9e17:b8b1]:2055"]

sh-4.4# /usr/bin/ovs-vsctl --timeout=15 -- --id=@sflow create sflow agent=ovn-k8s-mp0 targets='["[fe80::d060:9787:9e17:b8b1]:2055"]' -- set bridge br-int sflow=@sflow
9d2d9fb1-32f8-4f9b-b0fb-77fbded5164b
sh-4.4# ovs-vsctl find sflow
_uuid               : 9d2d9fb1-32f8-4f9b-b0fb-77fbded5164b
agent               : ovn-k8s-mp0
external_ids        : {}
header              : []
polling             : []
sampling            : []
targets             : ["[fe80::d060:9787:9e17:b8b1]:2055"]

```

Reply:
I guess we should add IPv6 for the whole feature.  Have to update the regex here:  https://github.com/openshift/api/blob/master/operator/v1/0000_70_cluster-network-operator_01.crd.yaml#L422   

I'll file a new BZ.